File: doc-examples/nitrite-example-java/src/main/java/example/WidgetRepository.java
package example;

import io.micronaut.data.nitrite.annotation.NitriteRepository;
import io.micronaut.data.repository.CrudRepository;

import java.util.List;
import java.util.UUID;

@NitriteRepository
public interface WidgetRepository extends CrudRepository<Widget, UUID> {
    List<Widget> findByName(String name);
}

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/WidgetRepository.kt
package example

import io.micronaut.data.nitrite.annotation.NitriteRepository
import io.micronaut.data.repository.CrudRepository
import java.util.UUID

@NitriteRepository
interface WidgetRepository : CrudRepository<Widget, UUID> {
    fun findByName(name: String): List<Widget>
}
